### DISCUSSION TITLE
Dedicated settings for scuba diving, underwater Mode

### DISCUSSION THREAD

**Discussion by nandodiver** [gopro/labs#1148] [Feature Requests]
I would like to see a dedicated preset for underwater Mode in Green water and blue water 

The basic settings for scuba diving for divers that want to maximize their gopro13 are the following: 

4K 60fps 
Lens Wide 
HLG HDR 
Depth 10 bit 
Bit rate High
Shutter Auto 
0EV 
ISO min 100 
ISO max 800 
Sharpness Medium 
Auto WB 
Color Natural 
Auto OFF never 
lcd OFF never

**Dive=1** Hypersmooth Autoboost
**WIDE=1** This offers a color gamut like setting white balance to Native (wider gamut), while enabling auto whitebalance
**BITR=150** - set the compression in Mb/s for the Protune High Bitrate setting (HEVC only). 

**Issues:** 
In HDR HLG Mode the ISO MAX is grey out at 1600, not possible to get ISO 800 and HLG at same time
**WBLK=1** Dont lock Wide balance totally  (ive tried underwater) 

**Requests underwater **
 Adjust Zoom for Macro
Auto Focus for Macro

**Comment by dnewman-gpsw** [comment]
$DIVE=1 (not Dive, case matters, you might know this, so this is for others)
$WBLK=1 Does lock Wide balance. What do you mean by totally?  It is designed to lock a second or so after capture start, and remains locked for the capture duration. 

HDR needs its ISO range to support the dual exposures it uses.

Auto focus, is not possible without a motor, you will need to use your fingers.
For Zoom, just use a standard lens profile with the Macro.

For Blue water try $WARM=50 (boost red 50%)
For Green water try $WARM=40$COOL=20 (boost red 40% and blue 20%)
The numbers will depend on the depth.

**Reply by nandodiver** [reply]
It seems using Dive=1 / WIDE=1 / BITR=150 affect all the other profiles, including GP-LOG 

I notice adding BITR=150 in profile made with QRcode, all the other existing profiles also get 150Mbit/s 

$DIVE=1 - optimizing the Hypersmooth stabilization for underwater. 
Ive tried hypersmooth - Autoboost vs ON and Autoboost has little edge, see the video below 

https://youtu.be/9LmZx0oGCdE?si=r7nIwxbZoxWqH_jC


$WBLK=1 Ive tried yesterday in Green water.... 
It indeed locks the White balance if theres artificial light reflect to the sensor (example filming something at close distance,  i see blue water background) , but if i move the camera ( capture duration wb lock) and theres no reflection of artificial light the Blue Water background starts to become green)

**Reply by nandodiver** [reply]
Here's whats i meant about the $WBLK=1 not locking the white balance on GP13

https://youtu.be/4mRncettKQc

Green water 
depth 25m 
2x dive lights 3750 lumens

Ive notice the background color change 
22 seconds Blue to Green 
27 seconds Green to Blue 
31 seconds blue to green 
36 seconds green to blue
47 seconds blue to green 
1m06 seconds green to blue 
1m55 seconds blue to green 

If the artificial light is not reflected to the sensor, the sensor just processed what he sees.
The color temperature doesn't lock

**Comment by dnewman-gpsw** [comment]
I wouldn't use autoboost, as if you move the camera too much, it ramps the lens distortion to linear, which isn't great underwater. Also the preview doesn't show this, so in autoboost LCD shows something slightly different that what is recorded. For minimal shake no problem, and underwater it is mostly a low shake environment.  So just an FYI.

I need to see the raw camera data for the second clip, either the MP4 or LRV, as that will have the metadata for what the white balance was doing.

**Reply by nandodiver** [reply]
Ive short the video to 1.7GB, link available for 7 days

**Reply by dnewman-gpsw** [reply]
It can't be a shorted, edited, or have any processing, as that strips out the metadata. If you want a smaller file, send the LRV.

**Reply by nandodiver** [reply]
Sorry i have LRV disable 
this one is smaller

**Reply by nandodiver** [reply]
and another one

**Reply by dnewman-gpsw** [reply]
Neither of the clips have WBLK or DIVE enabled.  You might be malforming your QR commands.  Please share the QR Code(s) you are setting.

You can see for yourself the header metadata (not the stream metadata) using https://gopro.github.io/labs/control/metadata/
![image](https://github.com/user-attachments/assets/7ab8d355-475e-4d5a-acfd-68f4633c2d2d)

WBLK and/or DIVE would be in this list if those feature where enabled. Even BITR and WIDE are missing.

**Reply by nandodiver** [reply]
i setup a different type of Qrcode for this weekend, i no longer have the QRCode 
I need to re-try again.

Thank you

**Reply by dnewman-gpsw** [reply]
Remember $DIVE=1 <- temporary, resets with a power off.  *DIVE=1 <- permanent, survives a power off.

**Reply by dnewman-gpsw** [reply]
Also please share the MISC/qrlog.txt file (and MISC/qrlog.bak) as those might still have the QR Codes you scanned.  These are a not large file, just logs of Labs actions.

**Reply by nandodiver** [reply]
where do i get does .txt files ? 

ive seen it...sd card

**Reply by dnewman-gpsw** [reply]
In the MISC folder

**Reply by nandodiver** [reply]
here the file

**Reply by dnewman-gpsw** [reply]
Lots of weirdness in there

!N$TOWB=1$NATU=1   <- what where you trying to do here?
!Nÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿ$PLOG=1$TIVO=1  <- and here?

Lots of illegal characters used. 

It seems you are going overboard with QRControl.  Don't combine setting (resolution and frame rate, etc.) with extensions (BITR, WIDE etc.  

These are the crazy command you ran (these are issue in that)
oMWBLK=1mHr4p60fWe4oSM0aAhS0dR0wAx0sMaLaAcNb1d1i8M1S0dVoV4oB7oW5oD2oCoSv0**null**q1oR1!MOWNR="xxxyyyzzz\nfernando\ncipreia dive club"oMBITR=150oMCOOL=20oMWARM=60

Set OWNR once in one command
```!MOWNR="xxxyyyzzz\nfernando\ncipreia dive club"```

Set you settings once (then save it a preset, so you not need to scan again)
```mHr4p60fWe4oSM0aAhS0dR0wAx0sMaLaAcNb1d1i8M1S0dVoV4oB7oW5oD2oCoSv0q1oR1```

The scan your extensions. You should make them permanent (so you scan only once.
```*WBLK=1*BITR=150*COOL=20*WARM=60```

Then after your dive undo with
```*WBLK=0*COOL=0*WARM=0```

**Reply by nandodiver** [reply]
Thank you

**Reply by nandodiver** [reply]
ive insert the sdcard and change the ownr
The latest information on txt is this 

[2024-11-03 19:30:00 31%   ] Labs:Main Start 
[2024-11-03 19:30:00 31%   ] Labs:H24.01.01.20.70 
[2024-11-03 19:30:00 31%   ] Labs:Initialize 
[2024-11-03 19:30:00 31%   ] preset CMD:!N$PLOG=1$TIVO=1 
[2024-11-03 19:30:03 31%   ] preset CMD:!N$PLOG=0$TIVO=0 
[2024-11-03 19:30:07 31%   ] Read QR:!MDIVE=1mHr4p60fWe3aAwAx0sMaLaAcNb1d1i8M1S0g0dVoV4oB4oW5oD2oCoSv0nullq1oR1!MBITR=150!MWIDE=1!MDIVE=1 
[2024-11-03 19:30:08 31%   ] Scans qrtxt:!MDIVE=1mHr4p60fWe3aAwAx0sMaLaAcNb1d1i8M1S0g0dVoV4oB4oW5oD2oCoSv0nullq1oR1!MBITR=150!MWIDE=1!MDIVE=1 
[2024-11-03 19:30:08 31%   ] Read QR:!MDIVE=1mHr4p60fWe3aAwAx0sMaLaAcNb1d1i8M1S0g0dVoV4oB4oW5oD2oCoSv0nullq1oR1!MBITR=150!MWIDE=1!MDIVE=1

**Reply by dnewman-gpsw** [reply]
Need to delete your preset called "$PLOG=1$TIVO=1"  

Need to clear everything from QRControl as you are still overloading, combine extensions and setting, and sending illegal commands (null is not a command.)

**Reply by nandodiver** [reply]
Need to reinstall FW....still having OWNR settings displaying when i turn the GP13  but i haven't set them on QRcode.

[2024-11-03 20:05:27 20%   ] Features activated: 
WIDE,WBLK,DIVE,BITR=150: 
[2024-11-03 20:05:27 20%   ] Labs:Main Start 
[2024-11-03 20:05:27 20%   ] Labs:H24.01.01.20.70 
[2024-11-03 20:05:27 20%   ] Labs:Initialize 
[2024-11-03 20:05:27 20%   ] preset CMD:!N$TOWB=1$NATU=1 
[2024-11-03 20:05:29 20%   ] Scans qrtxt:!MWBLK=1!MWIDE=1!MBITR=150 
[2024-11-03 20:05:29 20%   ] Read QR:!MWBLK=1!MWIDE=1!MBITR=150 
[2024-11-03 20:05:30 21%   ] Read QR:!MWBLK=1!MWIDE=1!MBITR=150

**Reply by dnewman-gpsw** [reply]
Reinstalling FW will do nothing.  Nothing is erased with a firmware update.  This is nothing wrong with the OWNR data you have.  There are only two ways to erase it:  Scan ```*OWNR=0``` or scan ```!RESET``` (then press the shutter three times quickly, this erases all Labs extensions.)

**Reply by nandodiver** [reply]
Ive reset Lab extensions, one question.
I have 2 profiles: 
1x  HLG/HDR 
1x GPLOG 

If i apply the QRcode only using the Extras Feature FourCC, does settings will be applied to does 2 profiles ?

**Reply by dnewman-gpsw** [reply]
I don't understand the question. Some extension shouldn't be combined with GPLog or HLG, like don't use WIDE=1, as both of these mode are wider gamut (less you want to mess with that.)

**Reply by nandodiver** [reply]
For underwater i want to use: 

Profile HLG/HDR
BITR=150 
WBLK=1

Profile HLG/HDR 
BITR=150 
AutoWB 

Profile HLG/HDR
BITR=150
WBLK=1
WIDE=1 

Profile GPLOG 
BITR=150

My question is if i create QRcode **only with extras** it will be applied to all existing Profiles when i scan  ? 
Or 
do i need to create **new profiles** completed with full QRcode  **mode, protune, prefs, extras**  ? if is the case, the existing Profiles will have **extras applied to them**  when i scan ?

**Comment by dnewman-gpsw** [comment]
As you shouldn't do HLG WIDE, you only rather have three profiles

1) Profile HLG/HDR
BITR=150
WBLK=1

2) Profile HLG/HDR
BITR=150
AutoWB

3) Profile GPLOG
BITR=150

All have BITR=150, so just set that globally.  Scan ```*BITR=150``` 
Now you only have 

1) Profile HLG/HDR
WBLK=1

2) Profile HLG/HDR  (stock)
AutoWB

3) Profile GPLOG (stock)

So you can create the two stock presets, with or without Labs.  The first one create with

![image](https://github.com/user-attachments/assets/85ad927e-f2c1-4148-a4cb-155488c93b03)

By calling the preset WBLK, it will only apply that extension when that preset is selected.

**Comment by JiDey0007** [comment]
I tried the WARM=99 and nothing obvious on two video tests with each other without

I think I did something wrong but it's hard to see what is really activated, not found in the metadata (sorry I've been using Labs for a while for Qr profiles but not yet perfected enough for the rest)

**Reply by dnewman-gpsw** [reply]
WARM=99 is not the full command, only ```$WARM=99``` or ```*WARM=99``` is. And the effect a 99% boost is instantly obvious, so you will know if you scan this correctly. All the fourCCs are precented with a action type, ```$``` for temporary or ```*``` for permanent.

**Reply by JiDey0007** [reply]
Great, thank you again! Everything works better with the complete code and I learned for temporary / permanent
